src/native/tls_context_options.c

```diff
@@ -125,7 +131,41 @@ jlong JNICALL Java_software_amazon_awssdk_crt_io_TlsContextOptions_tlsContextOpt
             aws_jni_throw_runtime_exception(env, "aws_tls_ctx_options_init_client_mtls_from_path failed");
             goto on_error;
         }
+    } else if (jni_pkcs11_options) {
+        tls->pkcs11_options = aws_tls_ctx_pkcs11_options_from_java_new(env, jni_pkcs11_options);
+        if (tls->pkcs11_options == NULL) {
+            /* exception already thrown */
+            goto on_error;
+        }
+
+        if (aws_tls_ctx_options_init_client_mtls_with_pkcs11(&tls->options, allocator, tls->pkcs11_options)) {
+            aws_jni_throw_runtime_exception(env, "aws_tls_ctx_options_init_client_mtls_with_pkcs11 failed");
+            goto on_error;
+        }
+    }
+#if defined(__APPLE__)
+    else if (jni_pkcs12_path && jni_pkcs12_password) {
```

Comment:
I moved this Apple/pkcs12 block of code up as a bug-fix. It used to be further down.

This fix ensures we do the `init()` calls before any of the `set()` calls on `aws_tls_ctx_options`.